tests/rules/no-effect-decorator.test.ts

```diff
@@ -11,194 +10,161 @@ import { ruleTester } from '../utils'
+    mapTo(CustomActions.pong()),
+  )}, { dispatch: false })
+}`,
       },
     ),
     fromFixture(
       stripIndents`
-      import { Injectable } from '@angular/core'
-      import { Effect } from '@ngrx/effects'
-      @Injectable()
-      export class FixtureEffects {
-        @Effect(config)
-        ~~~~~~~~~~~~~~~ [${noEffectDecorator}]
-        effect = this.actions.pipe(
-          ofType('PING'),
-          mapTo(CustomActions.pong()),
-        )
-        constructor(private actions: Actions) {}
-      }`,
+import { Effect } from '@ngrx/effects'
+
+class NotOk3 {
+  @Effect(config)
+  ~~~~~~~~~~~~~~~ [${noEffectDecorator}]
+  effect = this.actions.pipe(
+    ofType('PING'),
+    mapTo(CustomActions.pong()),
+  )
+}`,
       {
         output: stripIndents`
-        import { Injectable } from '@angular/core'
-        import { Effect, createEffect } from '@ngrx/effects'
-        @Injectable()
-        export class FixtureEffects {
+import { Effect, createEffect } from '@ngrx/effects'
 
-          effect = createEffect(() => { return this.actions.pipe(
-            ofType('PING'),
-            mapTo(CustomActions.pong()),
-          )}, config)
-          constructor(private actions: Actions) {}
-        }`,
+class NotOk3 {
+
+  effect = createEffect(() => { return this.actions.pipe(
+    ofType('PING'),
+    mapTo(CustomActions.pong()),
+  )}, config)
+}`,
       },
     ),
-    {
-      code: stripIndents`
-      import { Injectable } from '@angular/core'
-      import type { OnRunEffects } from '@ngrx/effects'
-      @Injectable()
-      export class FixtureEffects {
-        @Effect(config)
-        effect = this.actions.pipe(
-          ofType('PING'),
-          mapTo(CustomActions.pong()),
-        )
+    fromFixture(
+      stripIndents`
+import type { OnRunEffects } from '@ngrx/effects'
+
+class NotOk4 {
+  @Effect(config)
+  ~~~~~~~~~~~~~~~ [${noEffectDecorator}]
+  effect = this.actions.pipe(
+    ofType('PING'),
+    mapTo(CustomActions.pong()),
+  )
+}`,
+      {
+        output: stripIndents`
+import { createEffect } from '@ngrx/effects';
+import type { OnRunEffects } from '@ngrx/effects'
 
-        @Effect({ dispatch: false })
-        effect2 = createEffect(() => this.actions.pipe(
-          ofType('PING'),
-          mapTo(CustomActions.pong()),
-        ), config)
-        constructor(private actions: Actions) {}
-      }`,
-      output: stripIndents`
-      import { createEffect } from '@ngrx/effects';
-      import { Injectable } from '@angular/core'
-      import type { OnRunEffects } from '@ngrx/effects'
-      @Injectable()
-      export class FixtureEffects {
+class NotOk4 {
 
-        effect = createEffect(() => { return this.actions.pipe(
-          ofType('PING'),
-          mapTo(CustomActions.pong()),
-        )}, config)
+  effect = createEffect(() => { return this.actions.pipe(
+    ofType('PING'),
+    mapTo(CustomActions.pong()),
+  )}, config)
+}`,
+      },
+    ),
+    fromFixture(
```

Comment:
I had to separate this spec due to this: https://github.com/cartant/eslint-etc/issues/5.